.github/workflows/mac_release.yml

```diff
@@ -27,32 +27,72 @@ jobs:
     build:
         runs-on: macos-11
 
+        strategy:
+          fail-fast: false
+          matrix:
+            binary_compatibility:
+              - "darwin_16.x86_64"  # macOS 10.12, the earliest version supported by QT 5.13
```

Comment:
Do you think it's worth supporting 9-years old system? I'm asking this, cause this implies going back to Qt 5.13, whereas in SQLiteStudio 3.5 I plan to upgrade to Qt 6.5 and not looking back from then. I mean if someone will need to use SQLiteStudio on macOS 10.13 there will be SQLiteStudio 3.4.x, assuming the principle: the old OS, the old software. I mean I'm happy to support older systems, as long as it doesn't hold back the progress of the project. Sticking to supporting older Qt versions is holding back the progress.

Reply:
I would not be touching it if that problem did not hit me before (around version 3.3.0). I think there are many Mac, Linux, and Windows users that can't upgrade or won't upgrade for a lot of different reasons. I am mostly on MacOS 10.14 now, for example, but keep a 10.12 machine around. As for the support, no changes to the code are needed at the moment (ok, hung QtScript interpereters can't be terminated, but it's their own choice). Also, Qt 5.12 is specified as the minimum supported version somewhere in docs. Also, Windows 7 is somewhat 14 years old and is still supported by MinGW-W64, Qt 5 and SQLiteStudio.

Reply:
And... I think MacOS 10.12 users constitute around 8% of total MacOS users ;-) (source: https://ports.macports.org/port/curl/stats/ ; sadly there is no better data as Apple misreports at least versions 10.15+ on web stats)

Reply:
If the plan is to jump straight to 6.5, lots of old systems will be dropped. Qt 6.4 is at least 10.14+ compatible; Qt 6.5 is MacOS 11+ which makes it a no-go for me.

Reply:
If I may ask - why are you not upgrading to newer versions? I have a quite old Mac by myself (which I use very, very rarely) and like a year ago or so I upgraded to Monterey. For free and without issues.

I'm curious why to stick to older OS versions?

Reply:
I even googled what wise men answer to this question. The leading experts say: "Intellectual laziness, mostly".

Another way to call it is lack of time for foolishness: I know my tools well, though dated they are working well; for the new versions, I have to relearn, they are bloated, the new functionality they give (if any) is not useful for me, and often they break backward compatibility. The major OS version upgrades almost never go very smooth. OS X 10.9, Windows XP (in a VM) and Ubuntu 16.04 were good enough for me, the new versions never ran faster or increased backward compatibility; on the opposite, they break things (Linux systemd story), unconditionally promote my filesystem to another (slower and very young) filesystem type (MacOS 10.14+), or push unconditional updates (Windows 10+) and add a lot of telemetry (Windows 10+ and MacOS 11+).

![](https://i.kym-cdn.com/photos/images/newsfeed/001/044/247/297.png)

Reply:
6.2 has no compatibility advantages compared to 6.4, so 6.4 seems good enough for me.